src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,123 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private T[] arrayList;
+    private int minSize = 10;
+    private int numberOfElement = 0;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[minSize];
+
+    }
+
     @Override
     public void add(T value) {
-
+        arrayList[numberOfElement] = value;
+        numberOfElement++;
+        if (numberOfElement == arrayList.length) {
+            growArrayIfItIsFull();
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > numberOfElement || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "the index is not valid");
+        }
+        if (arrayList.length - numberOfElement == 0) {
+            growArrayIfItIsFull();
+        }
+        System.arraycopy(arrayList, index, arrayList, index + 1, numberOfElement - index);
+        arrayList[index] = value;
+        numberOfElement++;
     }
 
     @Override
     public void addAll(List<T> list) {
```

Comment:
Here you can simply loop through the list, and for each value use the ready-made add(value) method